utils.py

```diff
@@ -83,44 +88,102 @@ def MakePlots(truth_XY, predict_XY, PUPPI_XY, path_out):
-    bin_resolX_PUPPI, _, _                = binned_statistic(truth_PtPhi[:,0], truth_XY[:,0] - PUPPI_XY[:,0] * sfs_predict, statistic=resolqt, bins=binnings, range=(0,400))
-    bin_resolY_PUPPI, _, _                = binned_statistic(truth_PtPhi[:,0], truth_XY[:,1] - PUPPI_XY[:,1] * sfs_predict, statistic=resolqt, bins=binnings, range=(0,400))
+
+    def resolqt_upper(y):
+        return(np.percentile(y,90)-np.percentile(y,10))/2.0
+
+    def resolqt_lower(y):
+        return(np.percentile(y,76)-np.percentile(y,22))/2.0
+    
+    # response correction factors
+    sfs_ML = np.take(ml_means/truth_means,  np.digitize(true_ptPhi[:,0], binnings)-1, mode='clip')
+    sfs_PUPPI = np.take(puppi_means/truth_means, np.digitize(true_ptPhi[:,0], binnings)-1, mode='clip')
+
+    # compute resolutions inside each bin
+    bin_resolX_ML, bin_edges, binnumber = binned_statistic(true_ptPhi[:,0], trueXY[:,0] - mlXY[:,0] * sfs_ML, statistic=resolqt, bins=binnings, range=(0,400))
+    bin_resolY_ML, _, _ = binned_statistic(true_ptPhi[:,0], trueXY[:,1] - mlXY[:,1] * sfs_ML, statistic=resolqt, bins=binnings, range=(0,400))
+    bin_resolX_PUPPI, _, _ = binned_statistic(true_ptPhi[:,0], trueXY[:,0] - puppiXY[:,0] * sfs_PUPPI, statistic=resolqt, bins=binnings, range=(0,400))
+    bin_resolY_PUPPI, _, _ = binned_statistic(true_ptPhi[:,0], trueXY[:,1] - puppiXY[:,1] * sfs_PUPPI, statistic=resolqt, bins=binnings, range=(0,400))
+
+    # and the upper and lower bounds
+    bin_resolX_ML_upper, bin_edges, binnumber = binned_statistic(true_ptPhi[:,0], trueXY[:,0] - mlXY[:,0] * sfs_ML, statistic=resolqt_upper, bins=binnings, range=(0,400))
+    bin_resolY_ML_upper, _, _ = binned_statistic(true_ptPhi[:,0], trueXY[:,1] - mlXY[:,1] * sfs_ML, statistic=resolqt_upper, bins=binnings, range=(0,400))
+    bin_resolX_PUPPI_upper, _, _ = binned_statistic(true_ptPhi[:,0], trueXY[:,0] - puppiXY[:,0] * sfs_PUPPI, statistic=resolqt_upper, bins=binnings, range=(0,400))
+    bin_resolY_PUPPI_upper, _, _ = binned_statistic(true_ptPhi[:,0], trueXY[:,1] - puppiXY[:,1] * sfs_PUPPI, statistic=resolqt_upper, bins=binnings, range=(0,400))
+
+    bin_resolX_ML_lower, bin_edges, binnumber = binned_statistic(true_ptPhi[:,0], trueXY[:,0] - mlXY[:,0] * sfs_ML, statistic=resolqt_lower, bins=binnings, range=(0,400))
+    bin_resolY_ML_lower, _, _ = binned_statistic(true_ptPhi[:,0], trueXY[:,1] - mlXY[:,1] * sfs_ML, statistic=resolqt_lower, bins=binnings, range=(0,400))
+    bin_resolX_PUPPI_lower, _, _ = binned_statistic(true_ptPhi[:,0], trueXY[:,0] - puppiXY[:,0] * sfs_PUPPI, statistic=resolqt_lower, bins=binnings, range=(0,400))
+    bin_resolY_PUPPI_lower, _, _ = binned_statistic(true_ptPhi[:,0], trueXY[:,1] - puppiXY[:,1] * sfs_PUPPI, statistic=resolqt_lower, bins=binnings, range=(0,400))
+
+    # calculate ML and PUPPI resolutions weighted difference
```

Comment:
could you instead just get the resolution in a single large bin (rather than this weighted calculation?)

i think these should be equivalent and the other is perhaps more robust?

Reply:
Yep agreed, one bin is probably the way to do this!